### ISSUE TITLE
Crash webpack 5 at the end of compilation

### ISSUE THREAD

**Issue by CYB3RL1F3** [webpack/webpack#12265]
Hello,

I meet a crash with this error at the end of the compilation:
```
ERROR in [initial] main.e4e732f00dfe0da35af2.js
Maximum call stack size exceeded

webpack 5.11.0 compiled with 1 error in 38480 ms
```

Any clue?

I splitted lots of modules into small chunks.

Here is webpack config
COMMON :

```

module.exports = {
  context: sourcePath,
  mode,
  entry: {
    main: './main.tsx'
  },
  output: {
    path: outPath,
    filename: '[name].[fullhash].js',
    sourceMapFilename: '[name].[fullhash].map',
    chunkFilename: '[id].[fullhash].js',
    publicPath: '/'
  },
  target: 'web',
  resolve: {
    extensions: ['.js', '.ts', '.tsx'],
    mainFields: ['module', 'browser', 'main'],
    alias: {
      app: path.resolve(__dirname, 'src/app/'),
      types: path.resolve(__dirname, 'types/'),
      assets: path.resolve(__dirname, 'src/assets/')
    },
    fallback: {
      fs: false,
      net: false
    }
  },
  module: {
    rules: [
      // .ts, .tsx
      {
        test: /\.tsx?$/,
        use: isProduction
          ? ['ts-loader']
          : [
              {
                loader: 'babel-loader',
                options: {
                  plugins: ['react-hot-loader/babel']
                }
              },
              'ts-loader'
            ]
      },
      // css
      {
        test: /\.css$/,
        use: [
          {
            loader: MiniCssExtractPlugin.loader
          },
          {
            loader: 'css-loader',
            options: {
              fallback: 'style-loader',
              esModule: false,
              modules: {
                localIdentName: '[local]__[fullhash:base64:5]'
              },
              sourceMap: !isProduction,
              importLoaders: 1
            }
          },
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                ident: 'postcss',
                plugins: [
                  require('postcss-import')({ addDependencyTo: webpack }),
                  require('postcss-url')(),
                  require('postcss-cssnext')(),
                  require('postcss-reporter')(),
                  require('postcss-browser-reporter')({
                    disabled: isProduction
                  })
                ]
              }
            }
          }
        ],
        sideEffects: true,
      },
      // static assets
      { test: /\.html$/, use: 'html-loader' },
      { test: /\.png$/, use: 'url-loader?limit=10000' },
      { test: /\.jpg$/, use: 'file-loader' },
      { test: /\.svg$/, use: 'file-loader' },
      {
        test: /\.(webm|mp4)$/,
        use: {
          loader: 'file-loader',
          options: {
            name: 'videos/[name].[ext]'
          }
        }
      },
      {
        test: /\.(woff|woff2|eot|ttf|otf)$/,
        use: {
          loader: 'url-loader',
          options: {
            name: 'fonts/[name].[ext]',
            limit: 1
          }
        }
      }
    ]
  },
  optimization: {
    splitChunks: {
      cacheGroups: {
        commons: {
          chunks: 'initial',
          minChunks: 2
        },
        vendors: {
          test: /[\\/]node_modules[\\/]/,
          chunks: 'all',
          priority: -10
        }
      }
    },
  },
  plugins: [
    new FixStyleOnlyEntriesPlugin(),
    new CaseSensitivePathsPlugin(),
    new CleanWebpackPlugin(),
    new MiniCssExtractPlugin({
      filename: '[name].[fullhash].css',
      chunkFilename: '[name].[fullhash].css'
    }),
    new HtmlWebpackPlugin(index),
    new RobotstxtPlugin(robotsOptions),
    new SitemapWebpackPlugin({
      base: 'https://' + domain, 
      paths, 
      options: {
        fileName: 'sitemap.xml',
        lastMod: true,
        changeFreq: 'monthly',
        priority: 0.5
      }
    }),
    new webpack.DefinePlugin(
      configFromEnv
    ),
    /*
    new WorkboxPlugin.GenerateSW({
      clientsClaim: true,
      skipWaiting: true,
      maximumFileSizeToCacheInBytes: 20 * 1024 * 1024
    }),
    */
    new CopyWebpackPlugin({
      patterns: [
        { from: 'assets/pwa/manifest.json', to: 'pwa/manifest.json' },
        { from: 'assets/pwa/**/*', to: '[name].[ext]' },
        { from: 'assets/main.css', to: 'main.css' },
        { from: 'assets/fonts/**.*', to: 'fonts/[name].[ext]' },
        
      ],
    }),
    new WebpackPwaManifest(manifest),
    new WorkboxPlugin.InjectManifest({
      swSrc: manifest.serviceworker, // './assets/pwa/service_worker.js',
      swDest: 'service-worker.js'
    }),
    new FaviconsWebpackPlugin(
      path.resolve(__dirname, 'src/assets/pwa/favicon-96x96.png')
    )
  ],
  devServer: {
    contentBase: sourcePath,
    hot: true,
    inline: true,
    disableHostCheck: true,
    historyApiFallback: {
      disableDotRule: true
    },
    stats: 'minimal'
  },
  devtool: 'eval-source-map',
};

```

PROD :
```
merge(common, {
  mode: 'production',
  externals: {
    "mapbox-gl": "mapboxgl"
  },
  optimization: {
    minimizer: [terser],
    runtimeChunk: 'single',
    minimize: true,
    usedExports: true,
    providedExports: true,
    concatenateModules: true,
    moduleIds: 'deterministic',
    flagIncludedChunks: true,
    mergeDuplicateChunks: true,
    removeEmptyChunks: true,
    chunkIds: 'total-size', 
    moduleIds: 'size',
    splitChunks: {
      chunks: 'all',
      maxInitialRequests: 100,
      minSize: 0,
      cacheGroups: {
        common: {
          chunks: 'async'
        },
      'styled-components': {
          name: 'styled-components',
          chunks: 'async',
          filename: 'styled.components.[name].[fullhash].bundle.js',
          test: /[\\/]node_modules[\\/]styled-components[\\/]/
        },
        'react-html-parser': {
          name: 'react-html-parser',
          chunks: 'async',
          filename: 'html.parser.[name].[fullhash].bundle.js',
          test: /[\\/]node_modules[\\/]react-html-parser[\\/]/
        },
        node_modules: {
          name: 'node_modules',
          chunks: 'async',
          enforce: true,
          filename: '[name].[fullhash].bundle.js',
          test: /[\\/]node_modules[\\/](?!(@mapbox|mapbox-gl|styled-components|react-popupbox|dompurify|react-responsive|react-mapbox-gl|react-audio-player|react-html-parser|formik|date-fns|react|react-lottie|lottie-web|mobx-react|mobx-react-router|react-router|react-mobx|mobx|sanitize-html|react-backdrop-filter|@sentry|@sentry\/core|@sentry\/hub|@sentry\/minimal|@sentry\/browser)[\\/])/
        }
      }
    }
  },
  plugins: [
    terser,
    new webpack.DefinePlugin({
      'babel-plugin-styled-components': {
        pure: true
      }
    }),
    new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
    /*
    new BundleAnalyzerPlugin({
      analyzerPort: 8886,
    }),
    */
    new LodashModuleReplacementPlugin(),
    // new webpack.optimize.AggressiveMergingPlugin(),
    new CompressionPlugin({
      filename: '[name].[fullhash].gz[query]',
      algorithm: 'gzip',
      test: /\.js$|\.ts$|\.tsx$|\.css$|\.html$/,
      minRatio: 1
    })
  ]
});

```

Thanks a lot

**Comment by sokra** [comment]
hmm.. difficult to say. Do you have more details? Try `--stats-error-details` or `--stats-error-stack`

**Comment by CYB3RL1F3** [comment]
--stats-error-stack gives : 

```

EntrypointsOverSizeLimitWarning: entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  main (6.06 MiB)
      runtime.212175d7c1560dfdd270.js
      @sentry/core.212175d7c1560dfdd270.bundle.js
      @sentry/browser.212175d7c1560dfdd270.bundle.js
      react-dom.212175d7c1560dfdd270.bundle.js
      929.212175d7c1560dfdd270.js
      main.212175d7c1560dfdd270.js

    at compiler.hooks.afterEmit.tap.compilation (/Server/cyberlife/node_modules/webpack/lib/performance/SizeLimitsPlugin.js:139:7)
    at Hook.eval [as callAsync] (eval at create (/Server/cyberlife/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:14:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/Server/cyberlife/node_modules/webpack/node_modules/tapable/lib/Hook.js:18:14)
    at asyncLib.forEachLimit.err (/Server/cyberlife/node_modules/webpack/lib/Compiler.js:799:27)
    at /Server/cyberlife/node_modules/neo-async/async.js:2818:7
    at done (/Server/cyberlife/node_modules/neo-async/async.js:3522:9)
    at alreadyWritten (/Server/cyberlife/node_modules/webpack/lib/Compiler.js:645:8)
    at outputFileSystem.readFile (/Server/cyberlife/node_modules/webpack/lib/Compiler.js:730:19)
    at /Server/cyberlife/node_modules/graceful-fs/graceful-fs.js:123:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)

```



```

ERROR in [initial] main.212175d7c1560dfdd270.js
Maximum call stack size exceeded
ChunkRenderError: Maximum call stack size exceeded
    at errorAndCallback (/Server/cyberlife/node_modules/webpack/lib/Compilation.js:3250:26)
    at assetCacheItem.get (/Server/cyberlife/node_modules/webpack/lib/Compilation.js:3334:9)
    at hooks.get.callAsync (/Server/cyberlife/node_modules/webpack/lib/Cache.js:93:5)
    at Hook.eval [as callAsync] (eval at create (/Server/cyberlife/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Cache.get (/Server/cyberlife/node_modules/webpack/lib/Cache.js:75:18)
    at ItemCacheFacade.get (/Server/cyberlife/node_modules/webpack/lib/CacheFacade.js:116:15)
    at asyncLib.forEach (/Server/cyberlife/node_modules/webpack/lib/Compilation.js:3232:22)
    at arrayEach (/Server/cyberlife/node_modules/neo-async/async.js:2405:9)
    at Object.each (/Server/cyberlife/node_modules/neo-async/async.js:2846:9)
    at asyncLib.forEach (/Server/cyberlife/node_modules/webpack/lib/Compilation.js:3221:14)

```

Another thing : 

Until now it was working great with v4. Now, with v5, it ends with this, and when I run the app, it's really slow and bugging everywhere... I did updates for all dependencies, maybe it's not only webpack of course, but that's strange because no error in console or something, it's just... slow !!

A clue why with webpack 5 my chunks are much bigger and slow in runtime? :/

**Comment by alexander-akait** [comment]
> A clue why with webpack 5 my chunks are much bigger and slow in runtime? :/

Please don't mix questions in one issue, otherwise it will get confused and lose thread and you will not get solution

Please provide reproducible test repo, I think you are using incompatibility plugin

Also you ignore our the issue template, so don't expect a quick response, respect time of other developers

**Comment by Gladkiy-Alexey** [comment]
I have the same problem

**Comment by alexander-akait** [comment]
@Gladkiy-Alexey avoid spamming "same", please create reproducible test repo

**Comment by kpunith8** [comment]
What is the node version used? I had a similar issue migrating, updated to use, **14.5.0** it worked for me, give a try.

**Comment by CYB3RL1F3** [comment]
I was using the latest. Finally I found the cause, it was due to a plugin (I upgraded mini-css-extract-plugin and it conflicted with another one that was used to fix a problem with extract-text-plugin used before). 

It was not directly due to webpack, the real problem is the error was caught and impossible to find where it came from.

Meanwhile, I think you shouldn't catch those errors and let plugins crash.
I found by exploring the source code in node_modules...
